user-interface/src/main/java/life/qbic/datamanager/views/general/ComponentFunctions.java

```diff
@@ -0,0 +1,32 @@
+package life.qbic.datamanager.views.general;
+
+import static java.util.Objects.nonNull;
+
+import com.vaadin.flow.component.Component;
+
+public class ComponentFunctions {
+
+  private ComponentFunctions() {
+  }
+
+  /**
+   * Navigates up through the tree and checks whether the potential parent is an actual parent of
```

Comment:
```suggestion
   * Navigates up through the tree and checks whether the potential parent is an actual ancestor of
```
Since this iterates, I would rather call the relationship "ancestor", not parent